test/e2e/scenario/recorder/shadowDom.scenario.ts

```diff
@@ -0,0 +1,232 @@
+import type { RumInitConfiguration } from '@datadog/browser-rum-core'
+import { IncrementalSource, NodeType } from '@datadog/browser-rum/src/types'
+import type {
+  MouseInteractionData,
+  InputData,
+  SerializedNodeWithId,
+  TextNode,
+  BrowserMutationData,
+} from '@datadog/browser-rum/src/types'
+
+import {
+  findElementWithIdAttribute,
+  findElementWithTagName,
+  findFullSnapshot,
+  findIncrementalSnapshot,
+  findTextContent,
+} from '@datadog/browser-rum/test/utils'
+import type { EventRegistry } from '../../lib/framework'
+import { flushEvents, createTest, bundleSetup, html } from '../../lib/framework'
+import { browserExecute } from '../../lib/helpers/browser'
+
+/** Will generate the following HTML 
+ * ```html
+ * <my-open-web-component id="titi">
+ *  #shadow-root
+ *    <div for="div1-titi">
+ *      <label for="input-titi" id="label-titi">field titi: </label>
+ *      <input id="input-titi" value="toto">
+ *    </div>
+ *    <div>
+ *      <p id="text-titi">toto</p>
+ *    </div>
+ *</my-open-web-component>
+ *```
+ when called like `<my-open-web-component id="titi" />`
+ */
+const simpleShadowDom = `<script>
+class MyOpenWebComponent extends HTMLElement {
+  constructor() {
+    super();
+    this.attachShadow({ mode: "open" });
+  }
+  connectedCallback() {
+    const div2 = document.createElement("div");
+    div2.setAttribute("id", "div2-" + this.getAttribute("id"));
+    const p = document.createElement("p");
+    p.setAttribute("id", "text-" + this.getAttribute("id"));
+    p.innerText = 'toto'
+    div2.appendChild(p)
+    const div1 = document.createElement("div");
+    if (this.getAttribute("privacy")) {
+      div1.setAttribute("data-dd-privacy", this.getAttribute("privacy"));
+      div2.setAttribute("data-dd-privacy", this.getAttribute("privacy"));
+    }
+    const label = document.createElement("label");
+    label.setAttribute("for", "input-" + this.getAttribute("id"));
+    label.setAttribute("id", "label-" + this.getAttribute("id"));
+    label.innerText = "field " + this.getAttribute("id") + ": ";
+    const input = document.createElement("input");
+    input.setAttribute("id", "input-" + this.getAttribute("id"));
+    input.setAttribute("value", "toto");
+    input.addEventListener('input', (e) => {
+      p.textContent = e.target.value
+    })
+    div1.appendChild(label);
+    div1.appendChild(input);
+    div1.setAttribute("id", "div1-" + this.getAttribute("id"));
+    this.shadowRoot.appendChild(div1);
+    this.shadowRoot.appendChild(div2);
+  }
+}
+window.customElements.define("my-open-web-component", MyOpenWebComponent);
+</script>
```

Comment:
> maybe this could be mitigated by factorizing common parts with functions.

This seems fragile

Happy to discuss this over zoom because It's a bit of a pain to maintain, when write the e2e tests I had to have that in a HTML file on the side to check if it was working as expected, having multiple of them, albeit simpler, might complexify the whole process